django/apps/existing_database/query.py

```diff
@@ -0,0 +1,259 @@
+                    LEFT JOIN {UserGroupResult._meta.db_table} UGR USING (user_id)
+                WHERE timestamp::date >= (CURRENT_DATE- INTERVAL '30 days')
+                GROUP BY user_group_id, user_id, timestamp
+            )
+        SELECT
+            COUNT( DISTINCT user_group_id) as total_groups,
+            COUNT(DISTINCT user_id) as total_users,
+            SUM(swipe_count) as total_swipes
+        From dashboard_data
+    """
+    with connections[settings.MAPSWIPE_EXISTING_DB].cursor() as cursor:
+        cursor.execute(COMMUNITY_DAHBOARD_STATS_QUERY)
+        aggregate_results = cursor.fetchall()
+    for data in aggregate_results:
+        return CommunityStatsLatestType(
+            total_contributors_last_month=data[1] or 0,
+            total_groups_last_month=data[0] or 0,
+            total_swipes_last_month=data[2] or 0,
+        )
+
+
+def get_contributor_time_stats() -> List[ContributorTimeType]:
+    COMMUNITY_DASHBOARD_STATS_QUERY = f"""
+        WITH
+        user_data AS (
+            SELECT
+                MAX(R.start_time) as start_time,
+                MAX(R.end_time) as end_time,
+                R.timestamp as timestamp
+            From {Result._meta.db_table} R
+            GROUP BY timestamp
+        )
+    SELECT
+        timestamp::date as task_date,
+        SUM(
+            EXTRACT(
+                EPOCH FROM (end_time - start_time)
+            )
+        ) as total_time
+    From user_data
+    GROUP BY timestamp
+    """
+    with connections[settings.MAPSWIPE_EXISTING_DB].cursor() as cursor:
+        cursor.execute(COMMUNITY_DASHBOARD_STATS_QUERY)
+        aggregate_results = cursor.fetchall()
+    list_data = []
+    for data in aggregate_results:
+        list_data.append(
+            ContributorTimeType(total_time=round(data[1] / 60) or 0, task_date=data[0])
+        )
+    return list_data
+
+
+def get_project_type_query() -> List[ProjectTypeStats]:
+    PROJECT_TYPE_STATS_QUERY = f"""
+        SELECT
+            P.project_type as project_type,
+            SUM(st_area(P.geom::geography)) as area_sum
+            FROM {Project._meta.db_table} P
+                LEFT JOIN {Result._meta.db_table} R USING (project_id)
+            WHERE P.geom is not null AND P.project_type = '1'
+            GROUP BY P.project_type
+            """
+    with connections[settings.MAPSWIPE_EXISTING_DB].cursor() as cursor:
+        cursor.execute(PROJECT_TYPE_STATS_QUERY)
+        aggregate_results = cursor.fetchall()
+    project_list = []
+    for project_type, area_sum in aggregate_results:
+        project_list.append(
+            ProjectTypeStats(area=area_sum / 1000000, project_type=project_type)
+        )
+    return project_list
+
+
+def get_project_swipe_type() -> List[ProjectSwipeTypeStats]:
+    PROJECT_TYPE_STATS_QUERY = f"""
+        SELECT
+            P.project_type as project_type,
+            COUNT(*) as total_swipe
+            FROM {Result._meta.db_table} R
```

Comment:
I'm not sure if you can use a simple count here as the app interface is different for the project types. Here an overview how the number of results relates to the number of swipes:

- project type 1 and 4: 6 tasks --> 1 swipe
- project type 2 and 3: 1 task --> 1 swipe

Reply:
This has been changed. Can you maybe review this new file where we do this aggregation?
https://github.com/mapswipe/python-mapswipe-workers/blob/0d013617a60b747643ae21bd9b42fbff295fd687/django/apps/aggregated/management/commands/update_aggregated_data.py#L74-L83

**I will add project type 4 there as well.**
```sql
WHEN P.project_type in (1, 4) THEN ROUND(task_count/6)
```